test/unit/mdc-select/mdc-select-enhanced.test.js

```diff
@@ -0,0 +1,654 @@
+});
+
+test('adapter#addClass adds a class to the root element', () => {
+  const {component, fixture} = setupTest();
+  component.getDefaultFoundation().adapter_.addClass('foo');
+  assert.isTrue(fixture.classList.contains('foo'));
+});
+
+test('adapter#removeClass removes a class from the root element', () => {
+  const {component, fixture} = setupTest();
+  fixture.classList.add('foo');
+  component.getDefaultFoundation().adapter_.removeClass('foo');
+  assert.isFalse(fixture.classList.contains('foo'));
+});
+
+test('adapter#hasClass returns true if a class exists on the root element', () => {
+  const {component, fixture} = setupTest();
+  fixture.classList.add('foo');
+  assert.isTrue(component.getDefaultFoundation().adapter_.hasClass('foo'));
+});
+
+test('adapter_.floatLabel does not throw error if label does not exist', () => {
+  const fixture = bel`
+    <div class="mdc-select">
+      <div class="mdc-select__selected-text"></div>
+      <div class="mdc-select__menu mdc-menu mdc-menu-surface">
+      <ul class="mdc-list">
+        <li class="mdc-list-item mdc-list-item--selected"></li>
+        <li class="mdc-list-item" value="orange">          
+          Orange
+        </li>
+        <li class="mdc-list-item" value="apple">
+          Apple
+        </li> 
+      </ul>
+      </div>
+      <div class="mdc-line-ripple"></div>
+    </div>
+  `;
+  const component = new MDCSelect(fixture);
+  assert.doesNotThrow(
+    () => component.getDefaultFoundation().adapter_.floatLabel('foo'));
+});
+
+test('adapter.activateBottomLine and adapter.deactivateBottomLine ' +
+  'does not throw error if bottomLine does not exist', () => {
+  const fixture = bel`
+    <div class="mdc-select">
+      <div class="mdc-select__selected-text"></div>
+      <div class="mdc-select__menu mdc-menu mdc-menu-surface">
+      <ul class="mdc-list">
+        <li class="mdc-list-item mdc-list-item--selected"></li>
+        <li class="mdc-list-item" value="orange">          
+          Orange
+        </li>
+        <li class="mdc-list-item" value="apple">
+          Apple
+        </li> 
+      </ul>
+      </div>
+      <label class="mdc-floating-label">Pick a Food Group</label>
+    </div>
+  `;
+  const component = new MDCSelect(fixture);
+  assert.doesNotThrow(
+    () => component.getDefaultFoundation().adapter_.activateBottomLine());
+  assert.doesNotThrow(
+    () => component.getDefaultFoundation().adapter_.deactivateBottomLine());
+});
+
+
+test('#adapter.isRtl returns true when the root element is in an RTL context' +
+  'and false otherwise', () => {
+  const wrapper = bel`<div dir="rtl"></div>`;
+  const {fixture, component} = setupTest();
+  assert.isFalse(component.getDefaultFoundation().adapter_.isRtl());
+
+  wrapper.appendChild(fixture);
+  document.body.appendChild(wrapper);
+  assert.isTrue(component.getDefaultFoundation().adapter_.isRtl());
+
+  document.body.removeChild(wrapper);
+});
+
+test('instantiates ripple', function() {
```

Comment:
Move these above adapter tests so that the adapter tests above and below this are together?